src/v/lsm/core/lookup_result.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "base/format_to.h"
+#include "bytes/iobuf.h"
+
+#include <seastar/util/variant_utils.hh>
+
+#include <variant>
+
+namespace lsm {
+
+// The result of a `get` operation interally.
+class lookup_result {
+    struct missing_t {
+        bool operator==(const missing_t&) const = default;
+    };
+    struct tombstone_t {
+        bool operator==(const tombstone_t&) const = default;
+    };
+
+public:
+    static lookup_result missing() { return lookup_result(missing_t{}); }
+    static lookup_result tombstone() { return lookup_result(tombstone_t{}); }
+    static lookup_result value(iobuf b) { return lookup_result(std::move(b)); }
+
+    bool is_missing() const {
+        return std::holds_alternative<missing_t>(_value);
+    }
+    bool is_tombstone() const {
+        return std::holds_alternative<tombstone_t>(_value);
+    }
+    std::optional<iobuf> value() {
+        return ss::visit(
+          _value,
+          [](iobuf& buf) { return std::make_optional(buf.share()); },
+          [](const auto&) -> std::optional<iobuf> { return std::nullopt; });
+    }
+    std::optional<iobuf> take_value() {
```

Comment:
nit: mark this method with &&

Reply:
Done
813f52a75c7cf3f7214f44412538aae000ecc807